File: android/app/src/main/java/com/bwqr/mavinote/models/Device.kt
package com.bwqr.mavinote.models

import com.bwqr.mavinote.reax.Deserialize
import com.novi.serde.Deserializer

data class Device(
    val id: Int,
    val accountId: Int,
    val pubkey: String,
    val createdAt: String,
) {
    companion object : Deserialize<Device> {
        override fun deserialize(deserializer: Deserializer): Device {
            deserializer.increase_container_depth()

            val device = Device(
                deserializer.deserialize_i32(),
                deserializer.deserialize_i32(),
                deserializer.deserialize_str(),
                deserializer.deserialize_str(),
            )

            deserializer.decrease_container_depth()

            return device
        }
    }
}

File: ios/Mavinote/Models/Device.swift
import Serde

struct Device: Deserialize, Identifiable {
    let id: Int32
    let accountId: Int32
    let pubkey: String
    let createdAt: String

    static func deserialize(_ deserializer: Deserializer) throws -> Device {
        try deserializer.increase_container_depth()

        let device = Device(
            id: try deserializer.deserialize_i32(),
            accountId: try deserializer.deserialize_i32(),
            pubkey: try deserializer.deserialize_str(),
            createdAt: try deserializer.deserialize_str()
        )

        try deserializer.decrease_container_depth()

        return device
    }
}
